index.js

```diff
@@ -252,7 +252,7 @@ module.exports = {
     if (this.parent.isEmberCLIAddon()) {
       return this._findCoveredAddon().name;
     } else {
-      return this.parent.name;
+      return this.parent.name();
```

Comment:
Aha! This seems totally correct, sorry for the misleading comments without reading the diff carefully.